languages/ruby/dyp/lexer_ruby.mll

```diff
@@ -75,7 +75,7 @@ let emit_extra tok k = fun state lexbuf ->
 
 (* helper for transitioning between states *)
 let beg_choose want yes no = fun state lexbuf ->
-  if state.S.state == want
+  if phys_equal state.S.state want
   then yes state lexbuf
   else no state lexbuf
 
```

Comment:
Is this a semgrep bug? Should match `$X == $Y` from https://semgrep.dev/playground/r/ocaml.lang.correctness.physical-vs-structural.physical-equal

Reply:
It's because the source file is in ocamllex syntax, which is not supported by semgrep.